File: bubblesort/python/bubblesort.py
#!/usr/bin/python3


def bubblesort(tosort):
    found_sort = True
    while found_sort:
        found_sort = False
        for i,k in zip(range(len(tosort)-1), range(1,len(tosort))):
            if tosort[i] > tosort[k]:
                tosort[i], tosort[k] = tosort[k], tosort[i]
                found_sort = True

    return tosort

if __name__ == "__main__":
    assert(bubblesort([1,4,3,6,6,7,8,2,5,9,0]) == [0,1,2,3,4,5,6,6,7,8,9])

File: bubblesort/javascript/bubblesort.js
#!/usr/local/bin/node
var assert = require('assert');
var _ = require('underscore');

function main() {
    assert(_.isEqual(bubblesort([4,3,6,1,7,9,2,5,8]),[1,2,3,4,5,6,7,8,9]), "Not sorted correctly");

}


function bubblesort(tosort) {
    var didSwap = true;

    while (didSwap) {
        didSwap = false;
        for (var i=0; i < tosort.length-1; ++i) {
            if (tosort[i] > tosort[i+1]) {
                // pre es6 swapping else [a,b] = [b,a]
                tosort[i] = tosort[i+1] + (tosort[i+1] = tosort[i], 0);
                didSwap = true;
            }
        }
    }
    
    return tosort;
}



if (require.main === module) {
    main();
}
